numpy/_core/src/multiarray/array_assign_array.c

```diff
@@ -438,12 +443,17 @@ PyArray_AssignArray(PyArrayObject *dst, PyArrayObject *src,
         }
     }
 
+    int aligned =
+        copycast_isaligned(PyArray_NDIM(dst), PyArray_DIMS(dst), PyArray_DESCR(dst), PyArray_DATA(dst), PyArray_STRIDES(dst)) &&
+        copycast_isaligned(PyArray_NDIM(dst), PyArray_DIMS(dst), PyArray_DESCR(src), PyArray_DATA(src), src_strides);
+    int flags = (((NPY_SAME_VALUE_CASTING_FLAG & casting) != 0) << 1) | aligned;
```

Comment:
I think it makes a lot of sense to test for alignment here and pass that on, but it seems strange to change the value of the same-value casting flag. I'd tend to just pass on casting itself, with aligned just as a separate argument or included as a new very high value that (one hopes) is never going to be used (maybe listing it as  private via `_NPY_ALIGNED_CASTING` in the header files). 

(Obviously, no big deal, as it always can be done afterwards...)

Reply:
done